### PR TITLE
[Refactor] Further cleans of `csv_importer` 

### PR DESCRIPTION
**Pull request by dshemetov**
**Prerequisites**:

- [x] Unless it is a documentation hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### Summary

A follow up to #949. There were just some pieces of this code that really bothered me cause they were unnecessarily confusing to read every time I went there.

- Remove a bunch of unnecessary function arguments by correctly patching in tests
- Add type hints to many functions
- Simplify the logic that returns a list of CovidcastRows for database import
- Convert the path details tuple to a NamedTuples, so it's less of a mystery

### CODE REVIEW

File: src/acquisition/covidcast/csv_to_database.py
Comment (krivard) [inline src/acquisition/covidcast/csv_to_database.py:107]: praise: this is an enormous improvement

way more readable and less dependent on catching small errors in long tuple definitions. I have no idea why load_csv wasn't returning a CovidcastRow iterator to begin with.

File: tests/acquisition/covidcast/test_csv_importer.py
Comment (krivard) [inline tests/acquisition/covidcast/test_csv_importer.py:46]: question: why does this patch location need to be `delphi.epidata.acquisition.covidcast.csv_importer` when the patch for `isdir` on the line below can be in `os.path`?

File: tests/acquisition/covidcast/test_csv_to_database.py
Comment (krivard) [inline tests/acquisition/covidcast/test_csv_to_database.py:38]: nit: whitespace

```suggestion
    collect_files("fake_data_dir", False) # no specific issue
```

**Review by krivard — APPROVED** [review]
🏆 refactoring from `*_impl` to `@patch` has been on my irklist for ages

some questions and a suggestion but i wouldn't be mad if this merged as-is

experimentally using the Conventional Comments approach

File: src/acquisition/covidcast/csv_importer.py
Comment (dshemetov) [reply src/acquisition/covidcast/csv_importer.py:9]: So I'm using a Python import sort extension in my IDE and it likes to place bare `import X` statements above `from X import Y` statements, but then sorts each of those, so that's convention I followed here.

File: tests/acquisition/covidcast/test_csv_importer.py
Comment (dshemetov) [reply tests/acquisition/covidcast/test_csv_importer.py:46]: So the TLDR of this is due to Python imports
- `csv_importer` uses `import os`, so we can patch `os.path`, because `csv_importer` points to `os.path`
- `csv_importer` uses `from glob import glob`, which makes a new variable `glob` in the `csv_importer` namespace with the same value as `glob.glob`, but if we patch `glob.glob`, which updates that value to a Mock, `csv_importer.glob` retains its own value, so it's effectively not patched

So the end result is that you generally want to patch where a name is used, not a module itself.

[Here's a more complete blog post on this](https://nedbatchelder.com/blog/201908/why_your_mock_doesnt_work.html).

File: src/acquisition/covidcast/csv_importer.py
Comment (dshemetov) [reply src/acquisition/covidcast/csv_importer.py]: Yea, it's just a legacy ordering. I can change it to path ordering for now, though it may be something worth revisiting later.

File: src/acquisition/covidcast/csv_importer.py
Comment (dshemetov) [reply src/acquisition/covidcast/csv_importer.py:9]: I feel like this is an open ended question that we can address in a more comprehensive push for broader linting/formatting goals. Gonna keep as is for now.

**Comment by dshemetov** [comment]
Thanks @krivard ! Just going to merge this one in, we can add any further feedback to another issue.

**Comment by dshemetov** [comment]
Ok guess not, whenever @melange396 gets to it.

**Comment by krivard** [comment]
@dshemetov yeah I have it set to dismiss reviews when new commits are added; we had an intern that merged some unreviewed code a while back and it seemed prudent at the time. it's kindof a pain if we don't need it though. would you prefer to wait for George or want me to override and merge now?

**Comment by dshemetov** [comment]
I'd prefer to just merge it in, since George won't be back until Wednesday and I don't think this is that complicated of a change

**Pull request merged**
